examples/abitest/module_0/rust/tests/integration_test.rs

```diff
@@ -0,0 +1,197 @@
+        FRONTEND_MODULE_NAME,
+        FRONTEND_ENTRYPOINT_NAME,
+        ConfigMap::default(),
+    );
+    let runtime =
+        oak_runtime::configure_and_run(config).expect("unable to configure runtime with test wasm");
+
+    let (channel, interceptor) = oak_tests::channel_and_interceptor().await;
+    let client = OakAbiTestServiceClient::with_interceptor(channel, interceptor);
+
+    (runtime, client)
+}
+
+#[tokio::test(core_threads = 2)]
+#[serial]
+async fn test_abi() {
+    let (runtime, mut client) = setup().await;
+
+    // Skip tests that require the existence of an external service.
+    let mut req = AbiTestRequest::default();
+    req.exclude = "(Storage|GrpcClient|Roughtime)".to_string();
+
+    info!("Sending request: {:?}", req);
+    let result = client.run_tests(req).await;
+    assert_matches!(result, Ok(_));
+
+    info!("Runtime graph at exit is:\n{}", runtime.graph());
+
+    runtime.stop();
+
+    let mut disabled = 0;
+    let mut success = true;
+    let results = result.unwrap().into_inner().results;
+    for result in results {
+        if result.disabled {
+            disabled += 1;
+            continue;
+        }
+        info!(
+            "[ {} ] {}",
+            if result.success { " OK " } else { "FAIL" },
+            result.name
+        );
+        if !result.success {
+            error!("Failure details: {}", result.details);
+            success = false;
+        }
+    }
+    if disabled > 0 {
+        info!("YOU HAVE {} DISABLED TESTS", disabled);
+    }
+    assert_eq!(true, success);
+}
+
+#[tokio::test(core_threads = 2)]
+#[serial]
+async fn test_leaks() {
+    let (runtime, mut client) = setup().await;
+
+    let (before_nodes, before_channels) = runtime.object_counts();
+    info!(
+        "Counts before test: Nodes={}, Channels={}",
+        before_nodes, before_channels
+    );
+
+    // Run tests that are supposed to leave Node/channel counts in a known state.
+    let mut req = AbiTestRequest::default();
+    req.predictable_counts = true;
+
+    debug!("Sending request: {:?}", req);
+    let result = client.run_tests(req).await;
+    assert_matches!(result, Ok(_));
+    let results = result.unwrap().into_inner().results;
+
+    let (after_nodes, after_channels) = runtime.object_counts();
+    info!(
+        "Counts change from test: Nodes={} => {}, Channels={} => {}",
+        before_nodes, after_nodes, before_channels, after_channels
+    );
+
+    // Calculate the expected change in Node and channel counts for
+    // these tests.
+    let mut want_nodes = before_nodes as i32;
+    let mut want_channels = before_channels as i32;
+    for result in &results {
+        assert_eq!(result.predictable_counts, true);
+        if result.disabled {
+            continue;
+        }
+        want_nodes += result.node_change;
+        want_channels += result.channel_change;
+    }
+
+    if after_nodes != want_nodes as usize || after_channels != want_channels as usize {
```

Comment:
Can this if-block, or parts of it, be refactored into a separate method for readability?

Reply:
I don't think so &ndash; it would need all four parameters, so I don't think it would end up looking neater.  Getting rid of the casts would help though; I'll see if I can propagate some `usize`/`i32` change to help.

Reply:
Moved to `u32` throughout (need a sized type as values are passing through protobuf)